hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateDagVisitor.java

```diff
@@ -113,10 +121,40 @@ public Vertex onFilter(FilterPhysicalRel rel) {
                 ServiceFactories.nonSharedService(ctx ->
                         ExpressionUtil.filterFn(filter, SimpleExpressionEvalContext.from(ctx))),
                 (BiPredicateEx<Predicate<Object[]>, Object[]>) Predicate::test));
-        connectInput(rel.getInput(), vertex, null);
+        connectInputPreserveCollation(rel, vertex);
         return vertex;
     }
 
+    public Vertex onSort(SortPhysicalRel rel) {
+        ComparatorEx<?> comparator = ExpressionUtil.comparisonFn(rel.getCollations());
+
+        Vertex sortVertex = dag.newUniqueVertex("Sort",
+                ProcessorMetaSupplier.of(
+                        sortP(comparator)
+                )
+        );
```

Comment:
I would format these 4 lines on one:
```suggestion
                ProcessorMetaSupplier.of(sortP(comparator)));
```